### ISSUE TITLE
How to add another device that PiAlert didn't detect.

### ISSUE THREAD

**Issue by Just-another-pleb** [leiweibau/Pi.Alert#152]
I am very much at the bottom of the learning curve with how to use this.

There is an extra ROUTER that PiAlert didn't detect.

I am trying to use the `network` thing but am struggling how to get things connected.
I've looked at the FAQ and it goes over what the page is about but I am failing to see how to create connections between devices.

Thanks in advance.

**Comment by leiweibau** [comment]
Maybe this helps https://github.com/leiweibau/Pi.Alert/blob/main/docs/NETWORK_RELATIONSHIP.md

**Comment by leiweibau** [comment]
Please note that the network page is for visualization only. It has no further function of host detection. If the router is reachable from your network, you may be able to monitor its reachability via Web Service Monitoring or ICMP Monitoring.

**Comment by Just-another-pleb** [comment]
Thank you very much.

Yes, I appreciate it is only a visual picture and not controlled by what is actually detected.

I don't know if I am making my life any easier or not but I have been told I my network is somewhat *odd* (or *different*) to most.
Though I have VLANS, I also have MULTIPLE NAT in the network.

Quick attempt to show you what I mean:

--->internet<---->router 1<------>router 2<------->router 3<------->managed switch<-----------devices.

Router 1 is a *black box* that is not really doing anything.   It is merely an interface between the two types of cable.   So it doesn't count.

Router 2 is an *off the shelf* router that does NOT support VLANs.    It is there because if I call my provider's IT SUPPORT:  I have a base line to where they can help me.   If I didn't have this they would not want to help me as it would be too difficult for them.

Router 3 is my Edge router.   It does NAT.

PiHole only got as far as that.   It didn't see Router 2 - which ALSO DOES DO NAT!

It is academic but may be nice to have in my network picture.

I am kind of getting what is written in the link you showed me *just now*.
Although I had read it locally, the extra stuff on that page is helping.

**Comment by Just-another-pleb** [comment]
Um, I spoke too soon.

There is a huge disparity between what is shown in the link you provided and what I am seeing.

Somehow - in the link you gave me - the `internet` and `router` get connected but I am not able to do that.
As I have more important things to do just now I'll leave it, but would like to get that working if only to have a graphical representation of my network.

**Comment by leiweibau** [comment]
For simplicity:

If we assign directions like "left" and "right" to the hierarchy of your network, it would look like this in your case:

Internet -> Router 1 -> Router 2 -> Router 3 -> Switch - Hosts

The device "Internet" is the only device which is configured to the "right", because there is only one connection to your network.

For all other devices, the configuration on the details page of the device from "Port" and "Uplink Target" is a move to "left" (uplink).
At the switch, you specify router 3. For router 3, router 2, for router 2, 1 and for router 1, Internet. Since one can make however only one specification to the port per device, but routers have not only the Uplink, but also "Downlinks", there are limits here. The further problem is that router 1 and router 2 cannot be recognized at all, so they are not in the device list. Therefore, the manual port configuration is not available.
> ... **not appear in the devices list** (unmanaged switches) cannot be assigned to any other network component, not even via the "manual port configuration"

_I will make this clearer in the Documentation._

If you want to use the network page, I would recommend to you to do without the representation of Router 1 and 2, since these cannot be detected over the arpscan, anyway. So a configuration like:

Internet -> Router 3 -> Switch - Hosts

To monitor the reachability, you can use one of the other two possibilities, which I have implemented.

<hr>

> Oh, as I have VLANs, does PiAlert's graph support them?

No.

**Comment by Just-another-pleb** [comment]
Thanks very much for that.   I can live with what you showed me.

Do devices have to be ONLINE before I can use them in/on the `network` page?

I have devices that PiAlert knows about and they are off line.
PiAlert knows about them but they aren't showing in the Unassigned devices part.

**Comment by leiweibau** [comment]
> Do devices have to be ONLINE before I can use them in/on the network page?

No, they just need to be in the list. The status is irrelevant.

>I have devices that PiAlert knows about and they are off line.
PiAlert knows about them but they aren't showing in the Unassigned devices part.

Are these devices "Archived"? Archived devices are not displayed in "Unassigned devices". Once these devices have a port configuration, they appear where they should

**Comment by Just-another-pleb** [comment]
No, I haven't got to that stage of archiving things.  I am still setting up the network.

These are the two pages of the devices I see.

![Screenshot from 2023-08-23 16-48-32](https://github.com/leiweibau/Pi.Alert/assets/39888280/9494734c-1dd2-4106-8d7a-d1484f8247e0)

You see all the devices

On the network page this is what I have:


![Screenshot from 2023-08-23 16-48-44](https://github.com/leiweibau/Pi.Alert/assets/39888280/8fe5865a-3f48-4716-9e29-2b6162a9f35f)

As you can see, there are some devices not being shown.

**Comment by leiweibau** [comment]
For example, if you go to the details page of "TimePi", what does it say for "Uplink Target" and "Target Port Number"?

**Comment by Just-another-pleb** [comment]
Uplink Target -- 192.168.17.2  Port -- 2
But it isn't shown when I look at the managed switch.

(pics to help)
![Screenshot from 2023-08-23 16-59-47](https://github.com/leiweibau/Pi.Alert/assets/39888280/afbd54d9-a0b6-4cfc-bac4-fe39f75a542a)

![Screenshot from 2023-08-23 16-59-59](https://github.com/leiweibau/Pi.Alert/assets/39888280/58e63c7a-59b5-43a3-9adb-dda773a79efb)

**Comment by leiweibau** [comment]
Uplink Target must be selected via the Drop Down Menu on the right. The field references the IDs of the devices created on the network side

**Comment by leiweibau** [comment]
No problem. I am happy that I could help.
